vm/systemSmartContracts/staking.go

```diff
@@ -258,36 +258,8 @@ func (s *stakingSC) changeValidatorKey(args *vmcommon.ContractCallInput) vmcommo
 		s.eei.AddReturnMessage("changeValidatorKey function not allowed to be called by address " + string(args.CallerAddr))
 		return vmcommon.UserError
 	}
-	if len(args.Arguments) < 2 {
-		s.eei.AddReturnMessage(fmt.Sprintf("invalid number of arguments: expected min %d, got %d", 2, len(args.Arguments)))
-		return vmcommon.UserError
-	}
-
-	oldKey := args.Arguments[0]
-	newKey := args.Arguments[1]
-	if len(oldKey) != len(newKey) {
-		s.eei.AddReturnMessage("invalid bls key")
-		return vmcommon.UserError
-	}
-
-	stakedData, err := s.getOrCreateRegisteredData(oldKey)
-	if err != nil {
-		s.eei.AddReturnMessage("cannot get or create registered data: error " + err.Error())
-		return vmcommon.UserError
-	}
-	if len(stakedData.RewardAddress) == 0 {
-		// if not registered this is not an error
-		return vmcommon.Ok
-	}
-
-	s.eei.SetStorage(oldKey, nil)
-	err = s.saveStakingData(newKey, stakedData)
-	if err != nil {
-		s.eei.AddReturnMessage("cannot save staking data: error " + err.Error())
-		return vmcommon.UserError
-	}
-
-	return vmcommon.Ok
+	s.eei.AddReturnMessage("function is deprecated")
```

Comment:
Why not remove also L257-L260 if this function is deprecated to be called?

Reply:
As it exists right now in the mainnet, so it must stay for backward compatibility

Reply:
Ok